python/tests/visualization/test_bloch_sphere_visualization.py

```diff
@@ -0,0 +1,131 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+import os 
+import pytest
+import numpy as np
+
+import cudaq
+import qutip
+
+import io
+from contextlib import redirect_stdout
+
+
+@pytest.fixture(autouse=True)
+def do_something():
+    yield
+    cudaq.__clearKernelRegistries()
+
+# write sample kernels
+
+@cudaq.kernel
+def single_qubit_kernel():
+    # Run any random single qubit code. Currently using the code from examples/python/intro.py
+    qubit = cudaq.qubit()
+    h(qubit)
+    x(qubit)
+    y(qubit)
+    z(qubit)
+    t(qubit)
+    s(qubit)
+
+    mz(qubit)
+
+@cudaq.kernel
+def two_qubit_kernel(): 
+    # Run any random two-qubit code. Currently using the code from examples/python/expectation_values.py
+    qvector = cudaq.qvector(2)
+    x(qvector[0])
+    ry(np.random.default_rng().random*2*np.pi, qvector[1]) # random rotation angle between 0 and 2Pi
+    mz(qvector)
+
+
+# basic tests
+
+def test_visualization_bad_state():
+    with pytest.raises(Exception) as err:
+        cudaq.add_to_bloch_sphere(np.array([1,0]))
+
+def test_visualization_invalid_state():
+    with pytest.raises(Exception) as err:
+        cudaq.add_to_bloch_sphere(cudaq.get_state(two_qubit_kernel) )
+        
+
+def test_visualization_single_qubit_no_sphere():
+    b = cudaq.add_to_bloch_sphere(cudaq.get_state(single_qubit_kernel))
+    assert isinstance(b, qutip.Bloch)
+
+def test_visualization_single_qubit_shere():
+    sph = qutip.Bloch()
+    # generate a random density matrix with qutip and add to sphere
+    sph.add_states( qutip.rand_dm(2) )
+    
+    b = cudaq.add_to_bloch_sphere(cudaq.get_state(single_qubit_kernel),existing_sphere=sph)
+    assert isinstance(b, qutip.Bloch)
+
+
+# TODO: refactor this part later, but run the same tests as before with density matrix backend
+
+cudaq.set_target("density-matrix-cpu")
```

Comment:
This might affect the subsequent tests. Recommend to set / reset the target in the test fixture.

Reply:
Fixed in [a3964e2](https://github.com/NVIDIA/cuda-quantum/pull/1763/commits/a3964e284910f99cbbddddfa7493747ab9735855). I've added a line to reset the target after the  density matrix tests are completed.